File: 백준/Silver/10845. 큐/큐.py
import sys
input = sys.stdin.readline
queue = []

a = int(input())

for i in range(a):
    b = input().split()
    if(b[0] == "push"):
        queue.append(b[1])
    if(b[0] == "pop"):
        print(-1 if not queue else queue.pop(0))
    if(b[0] == "size"):
        print(len(queue))
    if(b[0] == "empty"):
        print(0 if queue else 1)
    if(b[0] == "front"):
        print(queue[0] if queue else -1)
    if(b[0] == "back"):
        print(queue[-1] if queue else -1)

File: 백준/Silver/10845. 큐/큐.js
console.log(
  solution(require("fs").readFileSync("/dev/stdin").toString().trim())
);

function solution(input) {
  const [_, ...items] = input.split("\n").map((i) => i.split(" "));
  let q = [];
  let r = [];

  items.forEach(([c, v]) => {
    switch (c) {
      case "push":
        q.push(v);
        break;
      case "pop":
        r.push(q.length ? q.shift() : -1);
        break;
      case "size":
        r.push(q.length);
        break;
      case "empty":
        r.push(q.length ? 0 : 1);
        break;
      case "front":
        r.push(q.length ? q[0] : -1);
        break;
      case "back":
        r.push(q.length ? q[q.length - 1] : -1);
        break;
    }
  });

  return r.join("\n");
}